src/NHibernate.Test/NHSpecificTest/NH2207/SampleTest.cs

```diff
@@ -9,10 +10,14 @@ namespace NHibernate.Test.NHSpecificTest.NH2207
 	[TestFixture]
 	public class SampleTest : BugTestCase
 	{
+		protected override bool AppliesTo(Dialect.Dialect dialect)
+		{
+			return dialect is MsSql2008Dialect;
+		}
+
 		protected override bool AppliesTo(ISessionFactoryImplementor factory)
 		{
 			return factory.ConnectionProvider.Driver is Sql2008ClientDriver;
-
```

Comment:
Indeed, this one is checked after schema creation. So if the schema cannot be created for the test, we have to test the driver by checking `TestCase.Cfg` property in the dialect `AppliesTo`, somewhat like [here](https://github.com/nhibernate/nhibernate-core/blob/master/src/NHibernate.Test/Pagination/CustomDialectFixture.cs#L33).

It is coded as if creating the factory was requiring the schema to be there.
```C#
Configure();
if (!AppliesTo(Dialect))
{
	Assert.Ignore(GetType() + " does not apply to " + Dialect);
}

CreateSchema();
try
{
	_sessionFactory = BuildSessionFactory();
	if (!AppliesTo(_sessionFactory))
	{
		Assert.Ignore(GetType() + " does not apply with the current session-factory configuration");
	}
}
catch
{
	DropSchema();
	throw;
}
```

We could rewrite it:

```C#
Configure();
if (!AppliesTo(Dialect))
{
	Assert.Ignore(GetType() + " does not apply to " + Dialect);
}

_sessionFactory = BuildSessionFactory();
if (!AppliesTo(_sessionFactory))
{
	Assert.Ignore(GetType() + " does not apply with the current session-factory configuration");
}
CreateSchema();
```